pkg/azure/aks/generate_regions.go

```diff
@@ -0,0 +1,196 @@
+// Code generation tool for Azure region mapping
+// This tool generates the mapClusterRegionToPricingRegion function
+// by fetching actual Azure regions and pricing API regions.
+
+//go:build generate
+
+package main
```

Comment:
Could we move the region generating tool to a separate package?

Reply:
good idea! moved to `pkg/azure/generate/`, done in next commit.